src/cpp/shared/game/game.cpp

```diff
@@ -268,130 +267,124 @@ void game::ProcessOnMapHeroes() {
   hero *randomHero;
   char mightBeHeroAlreadyExists[56];
   int ppMapExtraHeroIdx;
-  heroMapExtra *mapExtraHero;
+  HeroExtra *mapExtraHero;
   char isJail;
   int i;
 
-  result = memset(mightBeHeroAlreadyExists, 0, 54u);// Only 54 out of 56 bytes set to zero, for whatever reason.
-  for (i = 0; i < 3; ++i) {
-    for (coordYForRandomHero = 0; ; ++coordYForRandomHero) {
-      result = (void *)MAP_HEIGHT;
-      if (coordYForRandomHero >= MAP_HEIGHT)
-        break;
-      for (coordXForRandomHero = 0; coordXForRandomHero < MAP_WIDTH; ++coordXForRandomHero) {
-        loc = &this->map.tiles[(coordYForRandomHero * this->map.width) + coordXForRandomHero];
-        if ((loc->objType & 0x7F) == 55 || loc->objType == 251) {
-          isJail = (loc->objType & 0x7F) == LOCATION_JAIL;
-          ppMapExtraHeroIdx = loc->extraInfo;
-          mapExtraHero = (heroMapExtra *)ppMapExtra[ppMapExtraHeroIdx];
-
-          if (!i) {                           // 1st iteration
-            if (!mapExtraHero->field_11 || mapExtraHero->heroID >= 54 || mightBeHeroAlreadyExists[mapExtraHero->heroID]) {
-              mapExtraHero->couldBeHasFaction = 0;
-            } else {
-              mightBeHeroAlreadyExists[mapExtraHero->heroID] = 1;
-              mapExtraHero->couldBeHasFaction = 1;
-            }
-            if (isJail) {
-              mapExtraHero->owner = -1;
-            } else {
-              mapExtraHero->owner = loc->objectIndex / 7; // Uh, redundant line?
-              mapExtraHero->owner = gcColorToPlayerPos[mapExtraHero->owner];
-            }
+  memset(mightBeHeroAlreadyExists, 0, 54u);// Only 54 out of 56 bytes set to zero, for whatever reason.
+  for (coordYForRandomHero = 0; coordYForRandomHero < MAP_HEIGHT; ++coordYForRandomHero) {
```

Comment:
In general, variable names in the reverse-engineered code may be longer than you'd actually write in code, because (a) you never have to write them, and (b) we're still figuring out what the code does when you name them, and (c) in decompiled code, they always have scope to the full function.

Please make these names shorter (maybe just "y"), and restrict the scope to the body of the for loop. e.g.: "for (int y = 0; y < MAP_HEIGHT; y++)" .

Reply:
I've been working on trying to change some of these. I suppose the x/y names would be better in a simpler form. Also, I was under the impression that all of the variables should be declared at the beginning, but for iterators like that, I'll start practicing keeping their scopes where they really matter.

Reply:
"I was under the impression that all of the variables should be declared at the beginning"

Where'd you get that impression?

Reply:
This is just a pattern that I noticed in other functions and in the decompilations. I guess I never fully understood the convention.

Reply:
Nope, it's an artifact of the decompiler.